Update changelog for DAP-12. (#601)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

12:

- Bump draft-irtf-cfrg-vdaf-08 to 12 {{!VDAF}}, and specify the newly-defined
  application context string to be a concatenation of the DAP version in use
  with the task ID. (\*)

- Add support for "asynchronous" aggregation, based on the Leader polling the
  Helper for the result of each step of aggregation. (\*)

- Update collection semantics to match the new aggregation semantics introduced
  in support of asynchronous aggregation. (\*)

- Clarify the requirements around report replay protection, defining when and
  how report IDs must be checked and stored in order to correctly detect
  replays.

- Remove support for per-task HPKE configurations. (\*)

> - Remove the `max_batch_size` parameter of the fixed-size query type.

Reviewer: ```suggestion
- Remove the `max_batch_size` parameter of the "fixed-size" query type.
```

- Rename "query type" to "batch mode", to align the name of this configuration
  value with its functionality.

- Rename the "fixed-size" batch mode to "leader-selected", to align the name
  with the behavior of this query type.

- Restore the `part_batch_selector` field of the `Collection` structure, which
  was removed in draft 11, as it is required to decrypt collection results in
  some cases. (\*)

- Update `PrepareError` allocations in order to remove an unused value and to

>   reserve the zero value as unused. (\*)

Reviewer: ```suggestion
  reserve the zero value for testing. (\*)
```

- Document distributed-system and synchronization concerns in the operational
  considerations section.

- Document additional storage and runtime requirements in the operational
  considerations section.

> - Document deviations from the RFC 8446 presentation language for structures

Reviewer: ```suggestion
- Document deviations from the presentation language of {{Section 3 of !RFC8446}} for structures
```

  described in this specification.

> - Clarify that differential privacy mitigations will help with privacy, rather

Reviewer: ```suggestion
- Clarify that differential privacy mitigations can help with privacy, rather
```

12:

> - Bump draft-irtf-cfrg-vdaf-08 to 12 {{!VDAF}}. (\*)

Reviewer: Also mention the application context string and what value we set for it.

- Add support for "asynchronous" aggregation, based on the Leader polling the

>   Helper for completion. (\*)

Reviewer: ```suggestion
  Helper for the result of each step of the computation. (\*)
```

> - Update collection semantics to match the new aggregation semantics introduced

Reviewer: Is it the semantics that we're updating or just the "syntax"?

Author: I think I prefer "semantics" -- e.g. we're now indicating the current status of a collection job by an in-message status field, rather than using HTTP status. IMO, that's a semantic change.

  in support of asynchronous aggregation. (\*)

> - Clarify the requirements around report replay protection.

Reviewer: This refers to b99bcfd1c08786550147934640044af492cb9579, correct? Let's add a bit more context here. This bullet should answer the question: "do I need to change my implementation in some way?"

Author: I've updated the text a bit to add detail -- let me know what you think. I think implementations might need to change due to this -- e.g. Janus doesn't (I think), but it's plausible that there were other implementation strategies that no longer fit what is specified for replay protection with this change.

> - Remove the `max_batch_size` parameter of the fixed-size query type.

Reviewer: Is this a breaking change, i.e., do we need `(\*)`?

Author: I don't think it "breaks wire compatibility" -- `max_batch_size` is specified as one of the task parameters, which have no defined serialization & are negotiated out-of-band from DAP.

(This is very likely to be a wire-breaking change to downstream specs which do define serialization for the task parameters, however; e.g. taskprov will presumably need to drop this parameter.)

> - Rename the "fixed-size" query type to "leader-selected", to align the name

Reviewer: nit: I would swap the order of these: mention renaming first, and then say how the renamed thing has changed.
